schema/rules.go

```diff
@@ -1,19 +1,27 @@
+	ValidFieldNamePattern         = regexp.MustCompile(`^[a-zA-Z_$][a-zA-Z0-9_$]*$`)
 )
 
 var (
 	validators = []Validator{
 		&IndexSchemaValidator{},
 		&FieldSchemaValidator{},
+		&FieldNameValidator{},
```

Comment:
Actually, these validators are added for backward compatibility checks which means this code path won't execute when we receive a new schema. Add this check in the `Validate` method on the `fieldBuilder`. This `Validate` method is where we check if the field is valid and if we should accept the schema.

Reply:
Thanks for catching this Himank. After looking at the field.go - the build() method is more appropriate to place field name validation. Wdyt?

Reply:
Either place works, build or validate.

Reply:
updated. thanks!